### PR TITLE
Rspec views and fix n+1 problems

### PR DESCRIPTION
**Pull request by diegoh40**
- Write integration tests with Capybara gem.
- Secure app from n+1 problems.

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    it 'should return username' do
+      expect(page).to have_content('Diego')
+    end
+
+    it 'should return user posts' do
+      expect(page).to have_content('Number of Posts: 0')
+    end
+  end
+end
```
Comment (BertrandConxy) [inline spec/views/post_index_spec.rb:38]: - [ ] Dear Team, Referring to what the previous reviewer suggested in the Closed PR #2  It looks like there are still a lot of things that are not yet implemented. As requested by the previous reviewer please add the following tests:
 - Tests that assert you can see a post's title.
 - Tests that assert you can see some of the post's body.
 - Tests that assert you can see the first comments on a post.
 - Tests that assert you can see how many comments a post has.
 - Tests that assert you can see how many likes a post has.
 - Tests that assert you can see a section for pagination if there are more posts than fit on the view.
 - Tests that redirect to the post's show page when you click on a post.
 
- [ ] Also note that the `context` is wrong here you should write something meaningful that shows what you're testing, kindly adjust that 😃 🏃

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    it 'should return username' do
+      expect(page).to have_content('Diego')
+    end
+
+    it 'should return user posts' do
+      expect(page).to have_content('Number of Posts: 0')
+    end
+
```
Comment (BertrandConxy) [inline spec/views/post_index_spec.rb:25]: - [ ] It looks like these first three tests have been repeated. It is not a best practice to repeat things. We have to keep our code DRY which means not repeating code over and over again. As requested by the previous reviewer, please remove the redundant ones 😃 👍

File: spec/views/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  it 'shows the 1st comment on a post (too long!)' do
+    expect(page).to have_content('too long!')
+  end
+  it 'should return individual post' do    
+    expect(page).to have_content('Intro to JS')
+  end
+  it 'Not to show the Pagination button' do
+    expect(page).to have_button('Pagination')
+  end
+  it 'Redirect to post show page when a post is clicked' do
```
Comment (BertrandConxy) [inline spec/views/user_show_spec.rb:36]: - [ ] As requested by the previous reviewer in the Closed PR  #2  please add the following tests as they are in the project requirements 😃 👍 :
 - Tests that assert you can see the user's bio.
 - Tests that assert you can see the user's first 3 posts.
 - Tests that assert you can see a button that lets me view all of a user's posts.
 - Tests that redirect to the user's post's index page when you click on a **see all posts**.

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Highlights 💪 
- No linter errors
- correct flow is used
- good PR

⚠️ This Review has been done based on the previous reviewers' requested changes in the closed PR #2  Please next don't close and open a new PR to request another review, just request it directly on this current PR
 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi Team, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**